mteb/abstasks/AbsTaskClusteringFast.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+import logging
+import random
+from typing import Any, Dict, List
+
+import numpy as np
+import sklearn
+import sklearn.cluster
+from attr import dataclass
+from sklearn.metrics.cluster import v_measure_score
+
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+HFLang = str
+Split = str
+Sentences = List[str]
+ColumnName = str
+ColumnValues = List[str]
+HFDataset = Dict[Split, Dict[ColumnName, ColumnValues]]
+MultilingualDataset = Dict[HFLang, HFDataset]
+IsoLanguage = str
+ClusteringScores = List[Dict[str, float]]
+Scores = Dict[str, Any]
+
+
+def evaluate_clustering_bootstrapped(
+    embeddings: np.ndarray,
+    labels: list[str],
+    n_clusters: int,
+    cluster_size: int,
+    kmean_batch_size: int,
+    rng_state: random.Random = random.Random(),
+) -> ClusteringScores:
+    """Bootstrapped evaluation of clustering performance using V-measure.
+
+    The bootstrapping is done by sampling N samples from the corpus and clustering them. It is done without replacement to get a diverse set of
+    samples.
+    """
+    n_embeddings = embeddings.shape[0]
+    labels_arr = np.array(labels)
+
+    scores = []
+
+    clustering_model = sklearn.cluster.MiniBatchKMeans(
+        n_clusters=len(set(labels)),
+        batch_size=kmean_batch_size,
+        n_init="auto",
+    )
+
+    for _ in range(n_clusters):
+        # sample N samples from the corpus without replacement
+        cluster_indices = rng_state.sample(range(n_embeddings), k=cluster_size)
+
+        _embeddings = embeddings[cluster_indices]
+        _labels = labels_arr[cluster_indices]
+        clustering_model.fit(_embeddings)
+        cluster_assignment = clustering_model.labels_
+        v_measure = v_measure_score(_labels, cluster_assignment)
+        scores.append({"v_measure": v_measure})
```

Comment:
Could we append just `v_measure` instead of the dictionary, like in the normal case? Was there a special purpose to the dict?

Reply:
yeah it could just be a list instead

Reply:
It is now a list